cpp-ch/clickhouse.version

```diff
@@ -1,3 +1,3 @@
 CH_ORG=Kyligence
-CH_BRANCH=rebase_ch/20230809
-CH_COMMIT=461c7a0bf12
+CH_BRANCH=rebase_ch/20230816
```

Comment:
we already upgraded to 0817 see (https://github.com/oap-project/gluten/pull/2784)